File: Frontend/Kanban_Board_Major_Project/src/app/_models/project.ts
import { Stage } from "./stage";
import { User } from "./user";

export type Project = {
     projectId?: number;
     title?: string;
     description?: string;
     admin?: User;
     startDate?: Date;
     dueDate?: Date;
     priority?: string;
     completion?: number;
     stages?: Stage[];
     members?:User[];
 };
 

File: Backend/Kanban_Board_Major_Project/KanbanService/src/main/java/com/niit/kanban/KanbanService/domain/Project.java
package com.niit.kanban.KanbanService.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Document
public class Project {
    @Id
    private int projectId;
    private String title;
    private String description;
    private User admin;
    private Date startDate;
    private Date dueDate;
    private String priority;
    private double completion;
    private List<Stage> stages;
    private List<User> members;
}